docs/plugins.md

```diff
@@ -0,0 +1,138 @@
+
+## Plugin Repository
+
+In order to get plugins listed in the main argo rollouts documentation we ask that the plugin repository be created under
+the [argoproj-labs](https://github.com/argoproj-labs) organization. Please open an issue under argo-rollouts requesting a 
+repo which you would be granted admin access on. 
+
+There is also a standard naming convention for plugin names used for configmap registration, as well as what the plugin 
+uses for locating its specific configuration on rollout or analysis resources. The name needs to be in the form of 
+`<namespace>/<name>` and both <namespace> and <name> have a regular expression check that matches Github's requirements 
+for `username/org` and `repository name`. This requirement is in place to help with allowing multiple creators of the same plugin 
+types to exist such as `<org1>/nginx` and `<org2>/nginx`. These names could be based of the repo name such 
+as `argoproj-labs/rollouts-sample_prometheus-metric-plugin` but it is not a requirement. 
+
+There will also be a standard for naming repositories under argoproj-labs in the form of `rollouts-<tool>-<type>-plugin`
+where `<type>` is say `metric`, or `trafficrouter` and `<tool>` is the software the plugin is for say nginx.
+
+## Plugin Name
+
+So now that we have an idea on plugin naming and repository standards let's pick a name to use for the rest of this 
+documentation and call our plugin `argoproj-labs/nginx`.
+
+This name will be used in a few different spots the first is the config map that your plugin users will need to configure.
+It looks like this below.
+
+```yaml
+kind: ConfigMap
+metadata:
+  name: argo-rollouts-config
+data:
+  plugins: |-
+    metricProviders:
+    - name: "argoproj-labs/metrics"
+      location: "file:///Users/zaller/Development/argo-rollouts/metric-plugin"
+    trafficRouters:
+    - name: "argoproj-labs/nginx"
+      location: "file:///tmp/argo-rollouts/traffic-plugin"
+```
+
+As you can see there is a field called `name:` under both `metrics` or `trafficrouters` this is the first place where your
+end users will need to configure the name of the plugin. The second location is either in the rollout object or the analysis
+template which you can see the examples below.
+
+#### AnalysisTemplate Example
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: AnalysisTemplate
+metadata:
+  name: success-rate
+spec:
+  metrics:
+    - name: success-rate
+      ...
+      provider:
+        plugin:
+          argoproj-labs/metrics:
+            address: http://prometheus.local
+```
+
+#### Traffic Router Example
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Rollout
+metadata:
+  name: example-plugin-ro
+spec:
+  strategy:
+    canary:
+      canaryService: example-plugin-ro-canary-analysis
+      stableService: example-plugin-ro-stable-analysis
+      trafficRouting:
+        plugins:
+          argoproj-labs/nginx:
+            stableIngress: canary-demo
```

Comment:
It would be nice if the doc explains where `stableIngress` attribute comes from. Is this common field required for all traffic routers? Where the user can find the list of possible fields that can be defined for a plugin?

Reply:
This is explained right below, I cleaned up wordling slightly do you think it needs more than this.

``` You as a plugin author can then put any configuration you need under `argoproj-labs/nginx` and you will be able to
look up that config in your plugin via the plugin name key. You will also want to document what configuration options your plugin supports.```